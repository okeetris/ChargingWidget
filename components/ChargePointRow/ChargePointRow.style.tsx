import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    marginTop: 0,
    marginBottom: 2,
    backgroundColor: '#FFF',
    elevation: 2,
    borderStyle: 'solid',
  },
  'container:last-child': {
    borderRadius: 8,
    overflow: 'hidden',
    backgroundColor: '#F0F0F0',
  },
  containerDirections: {
    display: 'flex',
    flexDirection: 'column',
    backgroundColor: '#e6e1e1',
    padding: 'none',
    width: '50px',
    justifyContent: 'center',
    alignItems: 'center',
  },
  containerAvailable: {
    display: 'flex',
    flexDirection: 'column',
    padding: 'none',
    width: '50px',
    justifyContent: 'center',
    alignItems: 'center',
    height: 60,
    marginTop: 10,
    marginLeft: 10,
  },
  availText: {
    flex: 1,
    fontSize: 11,
  },
  title: {
    fontSize: 16,
    color: '#000',
  },
  containerText: {
    flex: 1,
    flexDirection: 'column',
    marginLeft: 12,
    justifyContent: 'center',
    padding: 10,
    width: '50%',
  },
  description: {
    fontSize: 11,
    marginTop: 4,
  },
  photo: {
    fontSize: 25,
    fontWeight: 'bold',
  },
});
