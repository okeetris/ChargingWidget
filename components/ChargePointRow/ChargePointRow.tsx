/* eslint-disable react/prop-types */
import React from 'react';
import { View, Text } from 'react-native';
import DirectionsIcon from '@mui/icons-material/Directions';
import StarIcon from '@mui/icons-material/Star';
import { kmConversion, availableConstructer } from '../../helpers/widgetHelper';
import { styles } from './ChargePointRow.style';

const ChargePointRow = ({ data }) => {
  return (
    <View style={styles.container}>
      <View style={styles.containerAvailable}>
        <Text style={styles.photo}>
          {availableConstructer(
            data.availableChargePoints,
            data.totalChargePoints,
          )}
        </Text>
        <Text style={styles.availText}>Available</Text>
      </View>
      <View style={styles.containerText}>
        <Text numberOfLines={1} style={styles.title}>
          {data.stationName}
        </Text>
        <Text style={styles.description}>{data.location}</Text>
      </View>
      {data.isFavourite ? (
        <StarIcon sx={{ color: 'gold', fontSize: '1rem' }} />
      ) : (
        <Text></Text>
      )}
      <View style={styles.containerDirections}>
        <DirectionsIcon sx={{ color: 'blue' }} />
        <Text>{kmConversion(data.distance)}</Text>
      </View>
    </View>
  );
};

export default ChargePointRow;
