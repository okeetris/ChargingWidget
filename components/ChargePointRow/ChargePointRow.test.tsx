import React from 'react';
import renderer from 'react-test-renderer';
import ChargePointRow from './ChargePointRow';

const testData = [
  {
    stationName: 'Test',
    dlocation: 'Test',
    availableChargePoints: 'Test',
    totalChargePoints: 'Test',
    isFavourite: 'Test',
    distance: 'Test',
  },
];
test('renders correctly', () => {
  const tree = renderer.create(<ChargePointRow data={testData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
