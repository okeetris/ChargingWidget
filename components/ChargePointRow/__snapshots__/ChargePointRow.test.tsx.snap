// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderStyle": "solid",
      "elevation": 2,
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 2,
      "marginTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "height": 60,
        "justifyContent": "center",
        "marginLeft": 10,
        "marginTop": 10,
        "padding": "none",
        "width": "50px",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    >
      undefined/undefined
    </Text>
    <Text
      style={
        Object {
          "flex": 1,
          "fontSize": 11,
        }
      }
    >
      Available
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginLeft": 12,
        "padding": 10,
        "width": "50%",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 11,
          "marginTop": 4,
        }
      }
    />
  </View>
  <Text />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e6e1e1",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": "none",
        "width": "50px",
      }
    }
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1jogx4x-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;color:blue;}",
        }
      }
      data-emotion="css 1jogx4x-MuiSvgIcon-root"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jogx4x-MuiSvgIcon-root"
      data-testid="DirectionsIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m21.71 11.29-9-9a.9959.9959 0 0 0-1.41 0l-9 9c-.39.39-.39 1.02 0 1.41l9 9c.39.39 1.02.39 1.41 0l9-9c.39-.38.39-1.01 0-1.41zM14 14.5V12h-4v3H8v-4c0-.55.45-1 1-1h5V7.5l3.5 3.5-3.5 3.5z"
      />
    </svg>
    <Text>
      undefinedm
    </Text>
  </View>
</View>
`;
